socorro/external/boto/crashstorage.py

```diff
@@ -272,6 +276,68 @@ class BotoS3CrashStorage(BotoCrashStorage):
 
 
 #==============================================================================
+class TelemetryBotoS3CrashStorage(BotoS3CrashStorage):
+    """S3 crash storage class for sending a subset of the processed crash
+    but reduced to only include the files in the processed crash
+    JSON Schema."""
+
+    required_config = Namespace()
+    required_config.resource_class = change_default(
+        BotoCrashStorage,
+        'resource_class',
+        'socorro.external.boto.connection_context.RegionalS3ConnectionContext'
+    )
+
+    required_config.elasticsearch = Namespace()
+    required_config.elasticsearch.add_option(
+        'elasticsearch_class',
+        default='socorro.external.es.connection_context.ConnectionContext',
+        from_string_converter=class_converter,
+        reference_value_from='resource.elasticsearch',
+    )
+
+    #--------------------------------------------------------------------------
+    def save_raw_and_processed(
+        self,
+        raw_crash,
+        dumps,
+        processed_crash,
+        crash_id
+    ):
+        all_fields = SuperSearchFields(config=self.config).get()
+        crash_report = {}
+
+        # Rename fields in raw_crash.
+        raw_fields_map = dict(
+            (x['in_database_name'], x['name'])
+            for x in all_fields.values()
+            if x['namespace'] == 'raw_crash'
+        )
+        for key, val in raw_crash.items():
+            crash_report[raw_fields_map.get(key, key)] = val
+
+        # Rename fields in processed_crash.
+        processed_fields_map = dict(
+            (x['in_database_name'], x['name'])
+            for x in all_fields.values()
+            if x['namespace'] == 'processed_crash'
+        )
+        for key, val in processed_crash.items():
+            crash_report[processed_fields_map.get(key, key)] = val
+
```

Comment:
Sure, that works too. I'm not sure it has a huge impact on performance though, so maybe it's not worth it, but do it if you think it's better. :)